src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,547 @@
+
+                        tmpevalfgrid = self._evaluate_free_pdf(ngrid,funcargs)
+                        
+                        # assign to list
+                        evalfgrid.append(tmpevalfgrid)
+
+
+                        for combo in combinations: 
+
+                            arrslice = []
+                            for griddim,entry in enumerate(combo):
+
+                                if entry==0:
+                                    arrslice.append(slice(0,self.gridshape[ngrid][griddim]-1))
+
+                                else: # must be a 1
+                                    arrslice.append(slice(1,self.gridshape[ngrid][griddim]))
+
+                            allcorners[ngrid].append(evalfgrid[ngrid][tuple(arrslice)].flatten())
+
+                    # now go back through and remake the corners to stack everything up
+                    corners = []
+                    for ncombo in range(0,len(combinations)):
+                        corners.append(np.hstack([allcorners[ngrid][ncombo] for ngrid in range(0,self.ngrids)]))
+
+
+            # do the sampling
+            X = _freesample(self.x0,self.x1,*corners,N_samples=N_samples,seed=self.rng)
+
+            return X
+
+
+        else:
+            raise ValueError("LintSampler.sample: eval_type is expected to be either gridsample or freesample.")
+
+        pass
+
+
+    def resetgrid(self,cells=()):
+        """Reset the sample grid without changing the pdf.
+        
+        Parameters
+        ----------
+        self : LintSampler
+            The LintSampler instance.
+
+        Returns
+        -------
+        combinations: numpy array
+            All possible combinations of starting and ending points for the
+            inferred dimensionality of the problem.
+                
+        """
+        self._setgrid(cells)
+
+
+    def _setgrid(self,cells):
+        """Construct the cells for sampling.
+
+        Parameters
+        ----------
+        self : LintSampler
+            The LintSampler instance.
+        cells : single array, tuple of arrays or list of tuples of arrays
+            If a single array, the boundary values for a 1D grid. If a tuple of arrays, the
+            boundary values for a grid with dimensionality of the length of the tuple. If
+            a list of tuples of arrays, the boundary values for an arbitrary number of
+            grids with dimensionality the length of the tuples.
+
+        Returns
+        -------
+        None
+        
+        Attributes
+        -------
+        eval_type : string
+            The evaluation type, either 'gridsample' or 'freesample'
+        dim : int
+            The inferred dimensionality of the problem
+        edgearrays : list of numpy arrays
+            The arrays defining the edge of the grids
+        edgedims   : list of integers, optional, used if eval_type == 'gridsample'
+            The shape of the gridsample grid
```

Comment:
Depending on the case, `edgedims` is sometimes an integer, sometimes a tuple, sometimes a numpy array. I make a code suggestion below to make it always a tuple. Here is an updated description to reflect this:
```suggestion
        edgedims   tuple of integers, optional, used if eval_type == 'gridsample'
            The shape of the gridsample grid
```